src/textEditor.ts

```diff
@@ -140,6 +141,32 @@ export class TextEditor {
     return vscode.window.activeTextEditor.document.getText(selection);
   }
 
+  /**
+   *  Retrieves the current word at position.
+   *  If current position is whitespace, selects the right-closest word
+   */
+  static getWord(position: Position) : string {
+    const whitespaceRegex = new RegExp("^ *$");
+    let start = position;
+    let end = position.getRight();
+
+    const char = TextEditor.getText(new vscode.Range(start, end));
+    if (whitespaceRegex.test(char)) {
+      start = position.getWordRight();
+    } else {
+      start = position.getWordLeft(true);
+    }
+    end = start.getCurrentWordEnd(true).getRight();
+
+    const word = TextEditor.getText(new vscode.Range(start, end));
+
+    if (whitespaceRegex.test(word)) {
+      throw VimError.fromCode(ErrorCode.E348);
```

Comment:
I see what you're saying, but hailing from other languages with optionals, IMO returning an empty optional is as declarative as throwing an error. I like it because now the return value indicates "this function can fail" and the caller is forced to deal with it, or else get type errors.

As it is, anyone else who calls this function will have to remember that it throws or else they'll get into trouble. 

I could see putting the status message on VimState actually. This seems nice to me because then we'd retain the behavior of updateView taking a VimState and rendering it to screen.

What do you think?